packages/next/src/build/webpack/loaders/next-app-loader.ts

```diff
@@ -293,27 +295,38 @@ async function createTreeCodeFromPath(
         continue
       }
 
+      // if the parallelSegment was not matched to the __PAGE__ slot, then it's a parallel route at this level.
+      // the code below recursively traverses the parallel slots directory to match the corresponding __PAGE__ for each parallel slot
+      // while also filling in layout/default/etc files into the loader tree at each segment level.
+
       const subSegmentPath = [...segments]
       if (parallelKey !== 'children') {
+        // A `children` parallel key should have already been processed in the above segment
+        // So we exclude it when constructing the subsegment path for the remaining segment levels
         subSegmentPath.push(parallelKey)
       }
 
-      const normalizedParallelSegments = Array.isArray(parallelSegment)
-        ? parallelSegment.slice(0, 1)
-        : [parallelSegment]
+      // if `parallelSegment` is an array, we grab
+      const normalizedParallelSegment = Array.isArray(parallelSegment)
+        ? parallelSegment[0]
+        : parallelSegment
 
-      subSegmentPath.push(
-        ...normalizedParallelSegments.filter(
-          (segment) =>
-            segment !== PAGE_SEGMENT && segment !== PARALLEL_CHILDREN_SEGMENT
-        )
-      )
```

Comment:
This was always operating on a single element and I'm not sure why it was being normalized into an array, so I refactored this.